src/document/DocumentAnnotator.ts

```diff
@@ -97,10 +114,21 @@ export default class DocumentAnnotator extends BaseAnnotator {
             return;
         }
 
-        if (mode === Mode.HIGHLIGHT) {
-            this.annotatedEl.classList.add('ba-is-highlighting');
-        } else {
-            this.annotatedEl.classList.remove('ba-is-highlighting');
+        ['ba-is-highlighting', 'ba-is-create-region'].forEach(
+            className => this.annotatedEl && this.annotatedEl.classList.remove(className),
```

Comment:
Is this existence check needed? It doesn't appear in the switch statement below.

Reply:
I was getting a flow issue earlier when I didn't have the existence check. Would it be better to wrap this logic as well as the switch below in an else case?

Reply:
Hm, this project is set up to use TypeScript rather than Flow. There is a guard clause a few lines up to handle `this.annotatedEl` being undefined. Maybe an IDE issue?

Reply:
I'm seeing this when I try to push a change that removes the existence check:

```src/document/DocumentAnnotator.ts:116:13 - error TS2532: Object is possibly 'undefined'.

116             this.annotatedEl.classList.remove(className),
                ~~~~~~~~~~~~~~~~

```

Reply:
```
const ANNOTATION_CLASSES: { [M in Mode]?: string } = {
    [Mode.HIGHLIGHT]: 'ba-create--highlight',
    [Mode.REGION]: 'ba-create--region',
};

// ... later ...
Object.values(ANNOTATION_CLASSES).forEach(className => {
    if (this.annotatedEl && className) {
        this.annotatedEl.classList.remove(className);
    }
});

this.annotatedEl.classList.add(ANNOTATION_CLASSES[mode] || '');
```